code/game/objects/items.dm

```diff
@@ -112,6 +112,10 @@
 	return ..()
 
 
+/obj/item/proc/update_item_state(mob/user)
+	item_state = "[initial(icon_state)][flags_item & WIELDED ? "_w" : ""]"
```

Comment:
you could throw in a check here for the `iconstate_w` icon_state existing
http://www.byond.com/docs/ref/#/proc/icon_states

Reply:
An automated check for non-existent icon states on init would be really good and show a lot of unmaintained objects. But doing a check every time an item state is updated or item wielded wouldn't make as much sense.